### PR TITLE
Add keywords to improve search in the class reference

### PR DESCRIPTION
**Pull request by Calinou**
- Follow-up to https://github.com/godotengine/godot/pull/78990.
- See also https://github.com/godotengine/godot-proposals/issues/7332.

### CODE REVIEW

File: doc/classes/Vector4.xml [hunk 1]
```diff
@@ -155,7 +155,7 @@
 				This method is faster than using [method is_equal_approx] with one value as a zero vector.
 			</description>
 		</method>
-		<method name="length" qualifiers="const">
+		<method name="length" qualifiers="const"  keywords="size" keywords="size">
```
Comment (Mickeon) [inline doc/classes/Vector4.xml:158]: Oops
```suggestion
		<method name="length" qualifiers="const"  keywords="size">
```

File: doc/classes/ReflectionProbe.xml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8" ?>
-<class name="ReflectionProbe" inherits="VisualInstance3D" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
+<class name="ReflectionProbe" inherits="VisualInstance3D" keywords="envmap" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
```
Comment (Mickeon) [inline doc/classes/ReflectionProbe.xml:2]: Hmmm...
```suggestion
<class name="ReflectionProbe" inherits="VisualInstance3D" keywords="envmap, environment" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
```

File: doc/classes/@GlobalScope.xml [hunk 3]
```diff
@@ -518,7 +518,7 @@
 				[/codeblocks]
 			</description>
 		</method>
-		<method name="inverse_lerp">
+		<method name="inverse_lerp" keywords="interpolate">
```
Comment (AThousandShips) [inline doc/classes/@GlobalScope.xml:521]: Add to `lerp` and `lerp_angle` too?

File: doc/classes/Light3D.xml [hunk 4]
```diff
@@ -88,12 +88,12 @@
 		<member name="light_negative" type="bool" setter="set_negative" getter="is_negative" default="false">
 			If [code]true[/code], the light's effect is reversed, darkening areas and casting bright shadows.
 		</member>
-		<member name="light_projector" type="Texture2D" setter="set_projector" getter="get_projector">
+		<member name="light_projector" type="Texture2D" setter="set_projector" getter="get_projector" keywords="cookie">
```
Comment (Mickeon) [inline doc/classes/Light3D.xml:91]: Legitimately curious now. If it wasn't me someone else would've asked later.

File: doc/classes/Vector4.xml (hunk 1 above)
Comment (AThousandShips) [reply doc/classes/Vector4.xml:158]:
```suggestion
		<method name="length" qualifiers="const" keywords="size">
```

File: doc/classes/Light3D.xml [hunk 5]
```diff
@@ -35,31 +35,31 @@
...
 		</member>
-		<member name="light_angular_distance" type="float" setter="set_param" getter="get_param" default="0.0">
+		<member name="light_angular_distance" type="float" setter="set_param" getter="get_param" default="0.0" keywords="pcss">
 			The light's angular size in degrees. Increasing this will make shadows softer at greater distances (also called percentage-closer soft shadows, or PCSS). Only available for [DirectionalLight3D]s. For reference, the Sun from the Earth is approximately [code]0.5[/code]. Increasing this value above [code]0.0[/code] for lights with shadows enabled will have a noticeable performance cost due to PCSS.
 			[b]Note:[/b] [member light_angular_distance] is not affected by [member Node3D.scale] (the light's scale or its parent's scale).
 			[b]Note:[/b] PCSS for directional lights is only supported in the Forward+ rendering method, not Mobile or Compatibility.
 		</member>
-		<member name="light_bake_mode" type="int" setter="set_bake_mode" getter="get_bake_mode" enum="Light3D.BakeMode" default="2">
+		<member name="light_bake_mode" type="int" setter="set_bake_mode" getter="get_bake_mode" enum="Light3D.BakeMode" default="2" keywords="gi_mode">
```
Comment (Mickeon) [inline doc/classes/Light3D.xml:62]: May as well say the full thing in that case? Or would that catch too many false positives?
```suggestion
		<member name="light_bake_mode" type="int" setter="set_bake_mode" getter="get_bake_mode" enum="Light3D.BakeMode" default="2" keywords="gi_mode, global_illumination">
```
With the way searches currently work (more than one term at once, I did that, yippie!!), searching "global illumination" will match.

**Review by Mickeon — COMMENTED** [review]
Apparently the diff is outdated but I will publish what I had in the meantime

File: doc/classes/AudioStreamPlayer.xml [hunk 6]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8" ?>
-<class name="AudioStreamPlayer" inherits="Node" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
+<class name="AudioStreamPlayer" inherits="Node" keywords="sound" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
```
Comment (Mickeon) [inline doc/classes/AudioStreamPlayer.xml:2]: Maybe? Very likely to use this for that purpose.
```suggestion
<class name="AudioStreamPlayer" inherits="Node" keywords="sound, music, song" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
```

File: doc/classes/AudioStreamPlayer2D.xml [hunk 7]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8" ?>
-<class name="AudioStreamPlayer2D" inherits="Node2D" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
+<class name="AudioStreamPlayer2D" inherits="Node2D" keywords="sound" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
```
Comment (Mickeon) [inline doc/classes/AudioStreamPlayer2D.xml:2]:
```suggestion
<class name="AudioStreamPlayer2D" inherits="Node2D" keywords="sound, sfx" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
```

File: doc/classes/AudioStreamPlayer3D.xml [hunk 8]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8" ?>
-<class name="AudioStreamPlayer3D" inherits="Node3D" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
+<class name="AudioStreamPlayer3D" inherits="Node3D" keywords="sound" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
```
Comment (Mickeon) [inline doc/classes/AudioStreamPlayer3D.xml:2]:
```suggestion
<class name="AudioStreamPlayer3D" inherits="Node3D" keywords="sound, sfx" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
```

File: doc/classes/Light3D.xml (hunk 4 above)
Comment (Calinou) [reply doc/classes/Light3D.xml:91]: Unity and some other 3D programs call them light cookies :slightly_smiling_face:

Comment (RedMser) [reply doc/classes/Light3D.xml:91]: Could add "gobo" as well. That's how Source engine calls it.

**Comment by RedMser** [comment]
Thank you so much for tackling this!
One request I have: I keep forgetting which of "editable, disabled, enabled and readonly" are used by which Controls, would be nice if all four keywords matched all related properties. At least later when the keywords are used to hint the gdscript auto complete, or filtering the inspector, it'll have more tangible benefits.

**Comment by Calinou** [comment]
> Thank you so much for tackling this! One request I have: I keep forgetting which of "editable, disabled, enabled and readonly" are used by which Controls, would be nice if all four keywords matched all related properties. At least later when the keywords are used to hint the gdscript auto complete, or filtering the inspector, it'll have more tangible benefits.

Done.

**Review by Mickeon — APPROVED** [review]
More can always be added later. This is a great start.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by Mickeon** [comment]
Caught you live 📸

**Pull request merged**